### PR TITLE
NRRDLoader handling of coronal and sagittal oriented scans.

### LINKED ISSUE #21898: NRRD volumes do not size slices properly when using NRRDs with a scan order different from axial.
The NRRD volumes are able to extract the image data properly, but the dimensions of the output are scaled improperly when the IJK to RAS matrix isn't the identity matrix. If the scan order of the NRRD isn't Axial [or rather, the axis are _not_ in the order of XYZ], then the images are stretched and transformed improperly.

### PR DESCRIPTION
**Pull request by developers-mirrorme3d**
Related issue: Fixed #21898 

**Description**

The IJK -> RAS transformation matrix is being loaded with hard-coded assumptions. The scans are presumed to fall under an (X, Y, Z) order. The transformation matrix is assumed to be the following.
```
[1,0,0]
[0,1,0]
[0,0,1]
```
This is true for any NRRD with an **axial** scan order, however, as a result of this assumption any **coronal** or **sagittal** oriented scan will be incorrectly transformed and look incredibly stretched on one axis and squished on another. The transformation matrix could be in the forms of:
```
Coronal:                Sagittal:
[1, 0, 0]               [0, 1, 0]
[0, 0, 1]               [0, 0, 1]
[0, 1, 0]               [1, 0, 0]
```

To illustrate this point, the header of an axial oriented NRRD is as follows, you will notice that the `space directions` field is in the form of the identity matrix and thus will be processed correctly by the NRRDLoader as it is now.

```
type: int
dimension: 3
space: left-posterior-superior
sizes: 512 512 149
space directions: (1.1972659999999999,0,0) (0,1.1972659999999999,0) (0,0,2.5168918918918921)
kinds: domain domain domain
endian: little
encoding: gzip
space origin: (-306.5,-306.5,-176.75)
```

An NRRD with a coronal scan order can have a header with `space directions` in a different orientation, which is where the NRRDLoader will struggle to transform properly:

```
type: int
dimension: 3
space: left-posterior-superior
sizes: 512 512 106
space directions: (0.48828100000000002,0,0) (0,0,-0.48828100000000002) (0,2,0)
kinds: domain domain domain
endian: little
encoding: gzip
space origin: (-129.63867200000004,-101.337891,147.570007)
```

The transformation matrix can also hold _negative_ values which indicate a flip, previously ignored and instead operated on by the `space` field within an NRRD. The proposed changes within this PR take this into account by the `space directions` matrix instead.

To summarize the proposed solution included in this PR:
- First identify the indices of each axis from reading the `space directions` matrix
- Re-order the reading of the matrix based on these axis.
- Use the `space directions` matrix as the source of truth for axis flipping instead of the `space` header entry

### CODE REVIEW

File: examples/jsm/loaders/NRRDLoader.js [hunk 1]
```diff
@@ -358,38 +358,32 @@ class NRRDLoader extends Loader {
...
+		volume.zLength = volume.dimensions[ zIndex ];
+
 		// spacing
-		const spacingX = ( new Vector3( headerObject.vectors[ 0 ][ 0 ], headerObject.vectors[ 0 ][ 1 ],
-			headerObject.vectors[ 0 ][ 2 ] ) ).length();
-		const spacingY = ( new Vector3( headerObject.vectors[ 1 ][ 0 ], headerObject.vectors[ 1 ][ 1 ],
-			headerObject.vectors[ 1 ][ 2 ] ) ).length();
-		const spacingZ = ( new Vector3( headerObject.vectors[ 2 ][ 0 ], headerObject.vectors[ 2 ][ 1 ],
-			headerObject.vectors[ 2 ][ 2 ] ) ).length();
+		const spacingX = ( new Vector3( headerObject.vectors[ xIndex ][ 0 ], headerObject.vectors[ xIndex ][ 1 ],
```
Comment (mrdoob) [inline examples/jsm/loaders/NRRDLoader.js:373]: I think these could be prettier:

```js
new Vector3().fromArray( headerObject.vectors[ xIndex ] ). length();
```

**Comment by developers-mirrorme3d** [comment]
The newest commit improves upon the first iteration of the solution by allowing for future extensions to [support different coordinate systems](http://teem.sourceforge.net/nrrd/format.html#space). For now, the solution fixes the NRRDLoader's initial ability to load both `left-posterior-superior` and `left-anterior-superior` coordinate systems by properly supporting axial, coronal, and sagittal oriented scans, whereas previously only axial scans properly loaded.

Before the suggested additions to the code, a sagittal scan would look similar to this:
brokenNRRDLoader

After the fix, they now scale properly.
fixedNRRDLoader

For the above examples, [I am using the following NRRD that was publically posted in a DICOM directory from the internet](https://www.dicomlibrary.com/?manage=02ef8f31ea86a45cfce6eb297c274598).

Some additional notes gathered from implementing this fix, for future thought:

- The NRRDLoader currently assumes that the output coordinate system is RAS, which is why there are transformations being done on the NRRDs to convert them from LPS/LAS -> RAS. This should be configurable in the future, as most medical systems use LPS as a default.
- The NRRDLoader only supports LPS/LAS coordinate systems, which can be extended to the other remaining ones in the future as needed.

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
@developers-mirrorme3d Made some clean up. Hopefully I didn't break anything. https://github.com/mrdoob/three.js/commit/2f6358dd55956e4f3595f304b2b2c925551228d3
